Newtemplate

rschroll opened the pull request:
This...got a little bit out of control.  I was implementing the new template, and then went to remove inputs that we didn't need.  That was easy enough, so I started changing the display of a couple of the elements.  That wasn't too bad, so I went to change things up to make it possible to add and remove items for arbitrary parts of lists.  Still not a huge issue, so I allowed reordering of skills.  And then drag-and-drop reordering of skill keywords.  That one was a pain, but it's finally working!

@Ampretuzo Please take a look to make sure we're not aiming at merge conflicts here.  If you want to review, go ahead, but this is mainly informational.

I want to take the reorderability of the skills page and implement it on the other pages.  It shouldn't be too bad, now that I've figured out this system.

Ampretuzo committed:
webpack devServer config proxying main website requests

Ampretuzo committed:
Our own copy of the splash screen

Ampretuzo committed:
Grab fellow key in HomeTdi

Ampretuzo committed:
initializeFellowJsonresume AsyncAction boilerplate

Ampretuzo committed:
Simplify fetch call

Ampretuzo committed:
Rename initialization action

Ampretuzo committed:
Save and Fetch buttons in SideNav

Ampretuzo committed:
saveFellowData

Ampretuzo committed:
Rename button: "Load Saved" => "Reset to Saved Values"

Ampretuzo committed:
Update the form on reset

Ampretuzo committed:
Fix tdi/actions.js

rschroll committed:
Remove template choice, and switch to TDI template as only choice

The mechanism is left in place on the server side, and the component
is left unused on the client side, should we want to reenable this
in the future.

The TDI template is currently just a copy of template6, but that will
be changed shortly.

rschroll committed:
Remove other templates' code

rschroll committed:
Implement TDI template

rschroll committed:
Remove the Awards section from client, and reorder remaining

The component is left in place, should we want to resurrect it.

rschroll committed:
Remove inputs not used by our template

rschroll committed:
Allow structured text in the project description

Multiple paragraphs and bulleted lists are possible now.  The sanitization
was stripping out any repeated whitespace, which removed doubled newlines.
I don't see why this was necessary, so it has been removed.

rschroll committed:
Make job responsibility entries into text areas

rschroll committed:
Add ability to add and remove individual skills and work details

rschroll committed:
Add ability to swap skill sections around

Currently this is done by clicking on the dividers between sections.

rschroll committed:
Move Row into common components

rschroll committed:
Add ability to remove individual skills

rschroll committed:
Allow drag-and-drop reordering of skill keywords

Right now this is directly in Skills, but it should be possible to
generalize to use in Jobs as well.

Ampretuzo committed:
Use redux-form initialValues to reset

Ampretuzo committed:
Fix initial empty json bug

Resetting to {} would lose a lot of properties that were used
by containers. That would cause a lot of undefined-related
errors.

Ampretuzo committed:
Don't initialize anything in HomeTdi

Ampretuzo committed:
Fix broken components after reset

Ampretuzo committed:
Remove unnecessary stuff

Ampretuzo committed:
Update tdi state (and therefore form initialValues) when the resume is saved

Ampretuzo reviewed (approved):
Impressive, Rob - that's a rockstar PR!
Let's merge this, I'll fix the conflicts if there are any.

Ampretuzo committed:
Collect our buttons in a container

Ampretuzo committed:
Admin initializtion

Ampretuzo committed:
Show error messages in publishPDF

Ampretuzo committed:
Let admins publish the PDF

rschroll committed:
Better dividers to allow swapping skill sections

rschroll committed:
Allow reordering and removing schools

rschroll committed:
Enable reordering and removing jobs

rschroll committed:
Allow swapping and removing of projects

rschroll committed:
Allow dragging of job responsibilities

rschroll committed:
Fix styling of various remove buttons

rschroll committed:
Remove progress bar in footer

rschroll committed:
Merge remote-tracking branch 'origin/integration-with-main-website' into newtemplate

rschroll committed:
Prettier website interaction buttons

The styling is taken from the buttons on the splash page, which, weirdly
enough, defines all its own elements.

rschroll committed:
Remove analytics code

Ampretuzo committed:
rm TmpButton.js

Ampretuzo committed:
Button "disabled" styling

Ampretuzo committed:
Disable publish button

Ampretuzo committed:
Install react-tooltip

NOTE: I'm not a fan of adding dependencies for such simple use cases - we should probably remove the tooltip.

Ampretuzo committed:
Tooltip and help text when Publish is disabled

Ampretuzo committed:
Single fetch function

Ampretuzo committed:
Initial fellow data load

Ampretuzo committed:
Store the key on initial admin load

Ampretuzo committed:
Less suggestive publish help wording

rschroll committed:
Improve styling of side bar

Ensure a fixed width that fits the buttons at the bottom.  Allow
vertical scrolling if there's not enough room.  Due to some strange
interaction between overflow and positioning, this requires another
div to be the thing that gets scroll bars.

rschroll committed:
Whitespace fixes

rschroll committed:
Always load state from the server when requested

It may have been changed by another user, or even the same user
in another browser.

rschroll committed:
Redirect standard users to the generator page.

The generator page then loads in the data from localstorage or the
server, as necessary.

rschroll committed:
Change logic around saving and publishing

There are three states to consider: the state of the form, the state
of the preview, and the saved state (at least as we last remember it).
Before, we allowed publishing when the preview state matched the saved
state.  But this produces odd configurations.  If the form state has
changed, but the preview is still the same as the saved state, hitting
"Save" causes removes the ability to publish (since it chanes the saved
state).  Or, if all three states are different, the user has to hit both
"Save" and "Preview" to be able to publish.

By autosaving on Publish, we can perform one of those syncs.  But the
PDF that we have corresponds to the preview state, which doesn't enter
into the save system.  In order to make sure the PDF corresponds to the
saved state, which is being set to the form state, we now only allow
publishing when the form state equals the preview state.  The publish
action (now "Save and Publish") will synchronize both the saved JSON
and PDF to this state.

I hope.

rschroll committed:
Use same button type for preview

Also, toggle the Preview and Save buttons alternatively.  Maybe this
will make things easier to understand?

rschroll committed:
Add Download and Upload buttons to the sidebar

This is a little tricky, since we want to make the download button
an <a> element, and <a> elements cannot be disabled.  So we swap
between an <a> and a disabled <button>.

The upload button isn't hooked up yet.

rschroll committed:
Add JSON upload from generator page

Ampretuzo committed:
Load preview on initialization

Ampretuzo committed:
Make admin behaviour more Fellow-ey

Ampretuzo committed:
Persist fellowKeyUrlsafe to local storage.

(So that the admin can refresh the page without losing
the ability to make requests to the main website.)

Ampretuzo committed:
Tweak initializeApplication.

So that the app will reset only when a Fellow key changes for admins.

Ampretuzo committed:
Remove resumake header.

Ampretuzo committed:
Toasts!

rschroll committed:
Add visa status

rschroll committed:
Slightly more robust templating

Don't try to include an itemize if there are no job highlights.
Avoid extra pipes in education section.

rschroll committed:
Add hyperlinks

rschroll committed:
Put up a progress screen when uploading or downloading data

Also, slight reworking of toast notifications.

rschroll committed:
Eliminate "phone mode"

Don't switch the layout until the width is below 50px.  Granted, it
gets harder to use the layout for narrow windows, but at least it
isn't broken.  We can consider fixing it up later.

rschroll committed:
Update about page, and add link from generator page

rschroll committed:
Clean up splash page

No one should end up here, but in case they do....

Ampretuzo committed:
Show errors a bit longer

Ampretuzo committed:
Fix initial load with undefined sections

Ampretuzo committed:
Separate form data and section order

Ampretuzo committed:
Disable save too when preview out of sync

rschroll committed:
Set initial state directly, rather than via a function

The latter throws an error on refreshing that the state is undefined.

rschroll committed:
Reset form values via JSON upload path

The form initialization path was putting the form in a weird state
where the onSubmit function was getting the initial values instead
of the current values as input.

rschroll committed:
Handle authentication errors from server

Errors get reported in two different JSON formats, depending on their
source.  Errors from handling the request directly get a list of error
strings.  Errors from being unauthenticated get a single error message.
We should probably settle on one format eventually, but the easier
solution is to be flexible on the client side.

rschroll committed:
Detect authorization errors and react appropriately

For 401 errors, offer a redirect to the login page.  For 403 errors,
just tell them they can't do that.

rschroll committed:
Await async functions

Also, window.location.path contains only the path, not the query
string, which is in .search.  Rather than divide up the new URL,
we'll just add it to the current domain.

rschroll committed:
Use proper variable name

rschroll committed:
Remove bit of whitespace at end of template

This can create a blank page 2 in some circumstances.

rschroll committed:
Control wrapping in the profile line(s)

Ideally, this will fit in one line.  But it if wraps multiple lines,
the line breaks occur only between components.  The pipe between
components only appears within a line.

rschroll committed:
Allow lines with left and right justified pieces to break between pieces

rschroll committed:
Allow LabeledInput to grow when inside of flex container

This keeps it from being too narrow.

rschroll committed:
Add mime types to nginx.conf

This ensures static files are served with the right types, so that
Firefox doesn't complain.

rschroll committed:
Change definition of fillorsplit

This uses \discretionary, rather than \linebreak[0], to keep LaTeX
from feeling as strongly about adding a line break.  The result is
that multi-line titles can still have the following element on the
last line, if there's space.

rschroll committed:
Fix link to instructions on profile page

rschroll committed:
Remove check_interval_sec on readiness_check

This was causing timeouts on upgrades, probably because it could only
try one or two checks before that timeout hit.  The default is 5
seconds, not 30, which should be fine here.
